blockdag/dag.go

```diff
@@ -162,6 +163,103 @@ type BlockDAG struct {
 	startTime                       mstime.Time
 }
 
+// New returns a BlockDAG instance using the provided configuration details.
+func New(config *Config) (*BlockDAG, error) {
+	// Enforce required config fields.
```

Comment:
I would argue that a databaseContext is also required. Please a nil check here. 
Alternatively, remove all the nil checks and let it panic.